### PR TITLE
add dev2 branch for RRFS data assimilation

### PR DESCRIPTION
**Pull request by hu5970**
Adde dev2 branch to hold the updates for Externals.cfg and scripts in src to build RRFS data assimilation cycles.

### CODE REVIEW

File: Externals.cfg [hunk 1]
```diff
@@ -43,5 +43,14 @@ hash = dceca26
 local_path = src/EMC_post
 required = True
 
+[GSI]
+protocol = git
+repo_url = https://github.com/hu5970/GSI.git
+# Specify either a branch name or a hash but not both.
+branch = feature/gsl_rrfs
```
Comment (christinaholtNOAA) [inline Externals.cfg:50]: This should be the feature/RRFS_dev2 branch.

**Review by christinaholtNOAA — CHANGES_REQUESTED** [review]

**Comment by hu5970** [comment]
Thanks for comments. Will resolve those comments when GSI and regional_workflow branches are ready.

File: Externals.cfg [hunk 2]
```diff
@@ -1,8 +1,8 @@
 [regional_workflow]
 protocol = git
-repo_url = https://github.com/NOAA-GSD/regional_workflow
+repo_url = https://github.com/hu5970/regional_workflow
```
[the remark this answers is not in the document]
Comment (hu5970) [reply Externals.cfg]: Done

File: src/build_gsi.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+if [ $USE_PREINST_LIBS = true ]; then
+  export MOD_PATH=/scratch3/NCEPDEV/nwprod/lib/modulefiles
+else
+  export MOD_PATH=${cwd}/lib/modulefiles
+fi
+
+# Check final exec folder exists
+if [ ! -d "../exec" ]; then
+  mkdir ../exec
+fi
```
[the remark this answers is not in the document]
Comment (hu5970) [reply src/build_gsi.sh:17]: Done

File: src/build_gsi.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+#  ./configure << EOT
+#4
+#EOT
+#  ./compile
+#elif [ "$target" = "gaea" ] ; then
+#    echo "Not doing anything for 'gaea', if statement reserved for future use"
+#elif [ "$target" = "odin" ] ; then
+#    ./compile odin
+#elif [ "$target" = "stampede" ] ; then
+#    ./compile stampede
```
[the remark this answers is not in the document]
Comment (hu5970) [reply src/build_gsi.sh:46]: Done

**Comment by hu5970** [comment]
@christinaholtNOAA Just updated the code based on your comments. Also, I added code to clone and compile refs_utl, which is for refs data preprocess and cloud/tten analysis. I will issue a PR for adding those components in regional_workflow soon.

File: src/build_gsi.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (christinaholtNOAA) [reply src/build_gsi.sh:17]: I am not seeing the change made here. There is no need for the if statement. The same is true for build_rrfs_utl.sh.

File: src/build_gsi.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+if [ ! -d "../exec" ]; then
+  mkdir -p ../exec
+fi
+
+cd gsi
+
+if [ "$target" = "jet" ] ; then
+  ./ush/build.comgsi
+elif [ "$target" = "hera" ] ; then
+  ./ush/build.comgsi
```
Comment (christinaholtNOAA) [inline src/build_gsi.sh:24]: Can this be an "or" expression so that we do not duplicate code? The same for build_rrfs_utl.sh.

**Review by christinaholtNOAA — APPROVED** [review]
A couple of minor changes listed below, but it looks good. We can merge once those changes are made.

File: src/install_all.sh [hunk 6]
```diff
@@ -92,6 +93,19 @@ $Build_gsi && {
  ${CP} regional_gsi.fd/exec/test_nc_unlimdims.x        ../exec/regional_test_nc_unlimdims.x
```
Comment (christinaholtNOAA) [inline src/install_all.sh:93]: The regional_gsi.fd/exec path does not exist when I build this code. Please ensure that this script will run for a fresh checkout of the code.

**Review by christinaholtNOAA — CHANGES_REQUESTED** [review]

File: src/build_gsi.sh (hunk 5 above)
Comment (hu5970) [reply src/build_gsi.sh:17]: Done

File: src/install_all.sh (hunk 6 above)
Comment (hu5970) [reply src/install_all.sh]: Comment out all the copies from regional_gsi.fd/exec.

This is from EMC GSI. Not what we are using now. But leave those items for future merging with EMC build.

File: src/build_gsi.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (hu5970) [reply src/build_gsi.sh]: delete this part. This script is not install the executable. There are a scripts for installation.

**Review by christinaholtNOAA — APPROVED** [review]

**Comment by hu5970** [comment]
@christinaholtNOAA If you have no more comments on this PR, please merge it. Thanks.

**Pull request merged**